pkg/auth/iam/iam.go

```diff
@@ -498,11 +508,17 @@ func respondError(httpStatus, errorCode int, errorMessage string) restful.Servic
 }
 
 // SetDevStackTracer used for activate verbose error message in non-prod environment
-// example usage : <package_alias>.SetDevStackTracer(configuration.Realm)
-func SetDevStackTracer(realm string) {
-	if realm != constant.DisabledDevstackTracerEnv {
-		DevStackTracer = true
+func SetDevStackTracer() error {
+	if realm, exists := os.LookupEnv("REALM_NAME"); exists {
+		if !exists {
+			return errors.New("unable to get REALM_NAME environment variable, DevStackTracer disabled")
+		}
+		if realm == constant.DisabledDevstackTracerEnv {
```

Comment:
for what reason to throw errors?

Reply:
I think we might need to log once it's not get the REALM_NAME.
But, I just realize that maybe better just log a warning directly inside `SetDevStackTracer` function. What do you think @angelo-chan ?